Spring Boot quickstarter ignores property no_nexus

**Describe the bug**
According to https://www.opendevstack.org/ods-documentation/ods-project-quickstarters/latest/be-java-springboot.html, it is possible to build the generated code without using nexus, by creating the `gradle.properties` file with content `no_nexus=true`.

**Expected behavior**
Local gradle build succeds without error.


**Affected version (please complete the following information):**
 - OpenShift: 3.11
 - OpenDevStack 2

**Log Output (ensure to remove any confidential information like tokens, project names, etc.**
```
D:\projekte\ods\demo\local\tst10-bespringboot>gradlew build
> Task :compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Failed to notify dependency resolution listener.
   > Could not resolve all dependencies for configuration ':detachedConfiguration1'.
      > Authentication scheme 'all'(Authentication) is not supported by protocol 'file'
   > Could not resolve all dependencies for configuration ':detachedConfiguration2'.
      > Authentication scheme 'all'(Authentication) is not supported by protocol 'file'

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
```

**Additional context**
Add any other context about the problem here.

--- a/be-java-springboot/postProcessProjectsettings.sh
+++ b/be-java-springboot/postProcessProjectsettings.sh
@@ -21,13 +21,13 @@ if [[ $version == "4.9" ]]; then
 	    nexus_user = "\${project.findProperty("nexus_user") ?: System.getenv("NEXUS_USERNAME")}"\
 	    nexus_pw = "\${project.findProperty("nexus_pw") ?: System.getenv("NEXUS_PASSWORD")}"\
 	' build.gradle
-	
+
 	sed -i.bak "s/\(apply plugin: 'java'\)/\1\napply plugin: 'maven'\napply plugin: 'jacoco'/g" build.gradle
-	
+
 	# by default no jar task in there .. we need to add it.
 	echo -e "bootJar {\n    archiveName    \"app.jar\"\n    destinationDir  file(\"\044buildDir/../docker\")\n}" >> build.gradle
 
-	# add nexus 
+	# add nexus
 	sed -i.bak 's/mavenCentral()/maven () {\
 	        url "${nexus_url}\/repository\/jcenter\/"\
 	        credentials {\
@@ -55,13 +55,13 @@ if [[ $version == "4.9" ]]; then
 	USE_LEGACY_NEXUS_UPLOAD_SCRIPT=1
 else
 	templateFile=../$SCRIPT_DIR/templates/build-$version.gradle
-	echo "using $templateFile" 
+	echo "using $templateFile"
 	# this allows quick config, new version - add new template, done
 	if [[ -f "$templateFile" ]]; then
 		echo "found specific gradle version template"
 		mv $templateFile build.gradle
 		USE_LEGACY_NEXUS_UPLOAD_SCRIPT=1
-	else 
+	else
 		# default
 		mv ../$SCRIPT_DIR/templates/build-4.10.gradle build.gradle
 		USE_LEGACY_NEXUS_UPLOAD_SCRIPT=0
@@ -69,7 +69,7 @@ else
 	sed -i.bak "s|__GROUP__|$GROUP|g" build.gradle
 fi
 
-rm build.gradle.bak	
+rm build.gradle.bak
 
 if [[ $USE_LEGACY_NEXUS_UPLOAD_SCRIPT == 1 ]]; then
   echo "add legacy nexus upload script to build.gradle"

--- a/be-java-springboot/postProcessProjectsettings.sh
+++ b/be-java-springboot/postProcessProjectsettings.sh
@@ -12,6 +12,8 @@ version=$(grep /gradle-. gradle/wrapper/gradle-wrapper.properties | cut -d "-" -
 
 echo "gradle version: $version"
 
+USE_LEGACY_NEXUS_UPLOAD_SCRIPT=0
+
 if [[ $version == "4.9" ]]; then
 	sed -i.bak '/springBootVersion =/a \
 	    nexus_url = "\${project.findProperty("nexus_url") ?: System.getenv("NEXUS_HOST")}"\
@@ -49,7 +51,8 @@ if [[ $version == "4.9" ]]; then
 	          password = "${nexus_pw}"\
 	        }\
 	      }\
-	/g' build.gradle	
+	/g' build.gradle
+	USE_LEGACY_NEXUS_UPLOAD_SCRIPT=1
 else
 	templateFile=../$SCRIPT_DIR/templates/build-$version.gradle
 	echo "using $templateFile" 
@@ -57,15 +60,19 @@ else
 	if [[ -f "$templateFile" ]]; then
 		echo "found specific gradle version template"
 		mv $templateFile build.gradle
+		USE_LEGACY_NEXUS_UPLOAD_SCRIPT=1
 	else 
 		# default
 		mv ../$SCRIPT_DIR/templates/build-4.10.gradle build.gradle
+		USE_LEGACY_NEXUS_UPLOAD_SCRIPT=0
 	fi
 	sed -i.bak "s|__GROUP__|$GROUP|g" build.gradle
 fi
 
 rm build.gradle.bak	
 
+if [[ $USE_LEGACY_NEXUS_UPLOAD_SCRIPT == 1 ]]; then
+  echo "add legacy nexus upload script to build.gradle"
 cat >> build.gradle <<EOL
 uploadArchives {
     repositories{
@@ -80,6 +87,9 @@ uploadArchives {
     }
 }
 EOL
+else
+  echo "do not add legacy nexus upload script to build.gradle"
+fi
 
 echo "fix nexus repo path"
 repo_path=$(echo "$GROUP" | tr . /)

--- a/be-java-springboot/templates/build-4.10.gradle
+++ b/be-java-springboot/templates/build-4.10.gradle
@@ -1,20 +1,33 @@
-plugins {
-	id 'org.asciidoctor.convert' version '1.5.3'
-	id 'org.springframework.boot' version '2.2.1.RELEASE'
-	id 'java'
-	id 'maven'
-	id 'jacoco'
+buildscript {
+    ext
+            {
+                nexus_url = "${project.findProperty('nexus_url') ?: System.getenv('NEXUS_HOST')}"
+                nexus_user = "${project.findProperty('nexus_user') ?: System.getenv('NEXUS_USERNAME')}"
+                nexus_pw = "${project.findProperty('nexus_pw') ?: System.getenv('NEXUS_PASSWORD')}"
+                no_nexus = (project.findProperty('no_nexus') ?: System.getenv('NO_NEXUS') ?: false).toBoolean()
+                if (!no_nexus && (nexus_url == "null" || nexus_user == "null" || nexus_pw == "null")) {
+                    throw new GradleException("property no_nexus='false' (or not defined) but at least one of the properties nexus_url, nexus_user or nexus_pw is not configured. Please configure those properties!")
+                }
+
+                def folderRel = (String) ("${project.findProperty('nexus_folder_releases') ?: System.getenv('NEXUS_FOLDER_RELEASES')}")
+                nexusFolderReleases = folderRel == "null" ? "maven-releases" : folderRel
+
+                def folderSnaps = (String) ("${project.findProperty('nexus_folder_snapshots') ?: System.getenv('NEXUS_FOLDER_SNAPSHOTS')}")
+                nexusFolderSnapshots = folderSnaps == "null" ? "maven-snapshots" : folderSnaps
+
+                snippetsDir = file('build/generated-snippets')
+            }
 }
 
-project.ext
-{
-		nexus_url =  "${project.findProperty('nexus_url') ?: System.getenv('NEXUS_HOST')}"
-		nexus_user =  "${project.findProperty('nexus_user') ?: System.getenv('NEXUS_USERNAME')}"
-		nexus_pw =  "${project.findProperty('nexus_pw') ?: System.getenv('NEXUS_PASSWORD')}"  
-        nexus_folder = "candidates"
-  
-        snippetsDir = file('build/generated-snippets')
+plugins {
+    id 'org.asciidoctor.convert' version '1.5.3'
+    id 'org.springframework.boot' version '2.2.1.RELEASE'
+    id 'java'
+    id 'maven'
+    id 'jacoco'
+    id 'maven-publish'
 }
+
 
 ext['spring-restdocs.version'] = '2.0.2.RELEASE'
 
@@ -25,58 +38,95 @@ version = '0.0.1-SNAPSHOT'
 sourceCompatibility = JavaVersion.VERSION_11
 
 repositories {
-	maven () {
-        url "${nexus_url}/repository/jcenter/"
-        credentials {
-          username = "${nexus_user}"
-          password = "${nexus_pw}"
+    if (no_nexus) {
+        println("using repositories 'jcenter' and 'mavenCentral', because property no_nexus=$no_nexus")
+        jcenter()
+        mavenCentral()
+    } else {
+        println("using nexus repositories")
+        maven() {
+            url "${nexus_url}/repository/jcenter/"
+            credentials {
+                username = "${nexus_user}"
+                password = "${nexus_pw}"
+            }
         }
-      }
 
-      maven () {
-        url "${nexus_url}/repository/maven-public/"
-        credentials {
-          username = "${nexus_user}"
-          password = "${nexus_pw}"
+        maven() {
+            url "${nexus_url}/repository/maven-public/"
+            credentials {
+                username = "${nexus_user}"
+                password = "${nexus_pw}"
+            }
         }
-      }
 
-      maven () {
-        url "${nexus_url}/repository/atlassian_public/"
-        credentials {
-          username = "${nexus_user}"
-          password = "${nexus_pw}"
+        maven() {
+            url "${nexus_url}/repository/atlassian_public/"
+            credentials {
+                username = "${nexus_user}"
+                password = "${nexus_pw}"
+            }
         }
-      }
-
+    }
 }
 
 dependencies {
-	implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
-	implementation 'org.springframework.boot:spring-boot-starter-data-rest'
-	implementation 'org.springframework.boot:spring-boot-starter-jersey'
-	implementation 'org.springframework.boot:spring-boot-starter-security'
-	implementation 'org.springframework.boot:spring-boot-starter-web'
-	compileOnly 'org.projectlombok:lombok'
-	runtimeOnly 'com.h2database:h2'
-	testImplementation 'org.springframework.boot:spring-boot-starter-test'
-	testImplementation 'org.springframework.restdocs:spring-restdocs-mockmvc'
-	testImplementation 'org.springframework.security:spring-security-test'
-  
+    implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
+    implementation 'org.springframework.boot:spring-boot-starter-data-rest'
+    implementation 'org.springframework.boot:spring-boot-starter-jersey'
+    implementation 'org.springframework.boot:spring-boot-starter-security'
+    implementation 'org.springframework.boot:spring-boot-starter-web'
+    compileOnly 'org.projectlombok:lombok'
+    runtimeOnly 'com.h2database:h2'
+    testImplementation 'org.springframework.boot:spring-boot-starter-test'
+    testImplementation 'org.springframework.restdocs:spring-restdocs-mockmvc'
+    testImplementation 'org.springframework.security:spring-security-test'
+
     testCompile "org.springframework.restdocs:spring-restdocs-webtestclient:${project.ext['spring-restdocs.version']}"
-    asciidoctor "org.springframework.restdocs:spring-restdocs-asciidoctor:${project.ext['spring-restdocs.version']}"  
+    asciidoctor "org.springframework.restdocs:spring-restdocs-asciidoctor:${project.ext['spring-restdocs.version']}"
 }
 
 test {
-	outputs.dir snippetsDir
+    outputs.dir snippetsDir
+    useJUnitPlatform()
 }
 
 asciidoctor {
-	inputs.dir snippetsDir
-	dependsOn test
+    inputs.dir snippetsDir
+    dependsOn test
 }
 
 bootJar {
-    archiveName    "app.jar"
-    destinationDir  file("$buildDir/../docker")
+    doLast {
+        println "Copy jar file build/libs/$bootJar.archiveName to docker directory!"
+        copy {
+            from "build/libs/$bootJar.archiveName"
+            into "$buildDir/../docker"
+            rename(bootJar.archiveName, "app.jar")
+        }
+    }
 }
+
+
+publishing {
+    repositories {
+        maven {
+            url = "${nexus_url}/repository/" + (version.endsWith('SNAPSHOT') ? nexusFolderSnapshots : nexusFolderReleases)
+
+            credentials {
+                username = "${nexus_user}"
+                password = "${nexus_pw}"
+            }
+
+        }
+    }
+    publications {
+        maven(MavenPublication) {
+            artifactId = rootProject.name
+            groupId = group
+            version = "${version}"
+            from components.java
+        }
+    }
+}
+

